### ISSUE TITLE
Macbook M1 CTRL+C (SIGTERM)  not working as expected

### ISSUE THREAD

**Issue by pollosp** [docker/compose#9100]
**Description**

Hi all , 
MacBook pro M1 CTRL+C (SIGTERM)  is not working as expected during a `docker-compose up` hitting ctrl+c , docker-compose doesn't return to the command prompt 

**Steps to reproduce the issue:**
1. We run `docker-compose up` for a project in Mac M1
2. Once containers are ready and healthy into the terminal with logs we send `CTRL+C`  (SIGTERM) 
3. Terminal where we send `CTRL+C`  starts to stopping containers

```
Running 8/8
 ⠿ Container A  Stopping                                                                                                                                        0.7s
 ⠿ Container B  Stopping                                                                                                                                       0.1s
```

4. Terminal where we send `CTRL+C`  set containers to  to stopped state
```
Running 8/8
 ⠿ Container A  Stopped                                                                                                                                   0.7s
 ⠿ Container B  Stopped                                                                                                                                    0.1s
``
5. Containers are stoped gracefully according to logs and `docker ps` command
6. Terminal where we send `CTRL+C` keep forever in stoped state not returning the terminal prompt 
7. If we send another `CTRL+C`  the get the following message and we end into the terminal prompt

```
```
ERRO[0058] got 3 SIGTERM/SIGINTs, forcing shutdown
[+] Running 0/0
 ⠿ Container A              Error while Killing                                                                                                                                 0.0s
 ⠿ Container B             Error while Killing
```

**Describe the results you received:**

Endless loop into stop state docker-compose insterface

**Describe the results you expected:**

After hitting `ctrl+c` in `docker-compose up` return to the terminal prompt after containers handle the SIGTERM and container are in stoped state

**Additional information you deem important (e.g. issue happens only occasionally):**

* All containers are stoped gracefully managing the SIGTERM according their logs , so no impact into the workload
* We tested this with different M1 machines, different compose projects and different , configuration like terms 
* `docker stop` and `docker-compose stop` work fine

**Output of `docker compose version`:**

```
# docker version
Client:
 Cloud integration: v1.0.22
 Version:           20.10.11
 API version:       1.41
 Go version:        go1.16.10
 Git commit:        dea9396
 Built:             Thu Nov 18 00:36:09 2021
 OS/Arch:           darwin/arm64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.11
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16.9
  Git commit:       847da18
  Built:            Thu Nov 18 00:34:44 2021
  OS/Arch:          linux/arm64
  Experimental:     false
 containerd:
  Version:          1.4.12
  GitCommit:        7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc:
  Version:          1.0.2
  GitCommit:        v1.0.2-0-g52b36a2
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
```
# docker-compose  version
Docker Compose version v2.2.1
```

**Output of `docker info`:**

```
docker info
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.7.1)
  compose: Docker Compose (Docker Inc., v2.2.1)
  scan: Docker Scan (Docker Inc., v0.11.0)

Server:
 Containers: 8
  Running: 0
  Paused: 0
  Stopped: 8
 Images: 10
 Server Version: 20.10.11
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 7b11cfaabd73bb80907dd23182b9347b4245eb5d
 runc version: v1.0.2-0-g52b36a2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.76-linuxkit
 Operating System: Docker Desktop
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 1.927GiB
 Name: docker-desktop
 ID: TPN4:BEHX:DN2Z:UWOI:ZIIW:ABJX:I3KF:6SZT:D7VH:OHT2:DUN2:MTDB
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by elamperti** [comment]
Having the same issue under Linux:

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc., v0.7.1-docker)
  compose: Docker Compose (Docker Inc., 2.2.2)

Server:
 Containers: 340
  Running: 0
  Paused: 0
  Stopped: 340
 Images: 192
 Server Version: 20.10.12
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: false
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1e5ef943eb76627a6d3b6de8cd1ef6537f393a71.m
 runc version: v1.0.3-0-gf46b6ba2
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.10.45-1-lts
 Operating System: Manjaro Linux
 OSType: linux
 Architecture: x86_64
 CPUs: 24
 Total Memory: 31.33GiB
 Name: irrelevant
 ID: V7AU:XCVG:6BTM:H5H3:RNEK:ROSX:U2RF:ZMW5:XLQX:PXEW:H6TU:KVZR
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```

**Comment by elamperti** [comment]
This is a duplicate of #8523 and it should be resolved with compose version 2.2.3 which was released some days ago (see PR #9019).
